src/main/scala/gitbucket/core/controller/ApiController.scala

```diff
@@ -54,15 +54,104 @@ trait ApiControllerBase extends ControllerBase {
     with CollaboratorsAuthenticator =>
 
   /**
+    * https://developer.github.com/v3/#root-endpoint
+    */
+  get("/api/v3/") {
+    JsonFormat(ApiEndPoint())
+  }
+
+  /**
+    * https://developer.github.com/v3/orgs/#get-an-organization
+    */
+  get("/api/v3/orgs/:groupName") {
+    getAccountByUserName(params("groupName")).filter(account => account.isGroupAccount).map { account =>
+      JsonFormat(ApiUser(account))
+    } getOrElse NotFound
+  }
+
+  /**
    * https://developer.github.com/v3/users/#get-a-single-user
    */
   get("/api/v3/users/:userName") {
-    getAccountByUserName(params("userName")).map { account =>
+    getAccountByUserName(params("userName")).filterNot(account => account.isGroupAccount).map { account =>
       JsonFormat(ApiUser(account))
     } getOrElse NotFound
   }
 
   /**
+    * https://developer.github.com/v3/repos/#list-organization-repositories
+    */
+  get("/api/v3/orgs/:orgName/repos") {
+    JsonFormat(getVisibleRepositories(context.loginAccount, Some(params("orgName"))).map{ r => ApiRepository(r, getAccountByUserName(r.owner).get)})
+  }
+  /**
+   * https://developer.github.com/v3/repos/#list-user-repositories
+   */
+  get("/api/v3/users/:userName/repos") {
+    JsonFormat(getVisibleRepositories(context.loginAccount, Some(params("userName"))).map{ r => ApiRepository(r, getAccountByUserName(r.owner).get)})
+  }
+
+  /*
+   * https://developer.github.com/v3/repos/branches/#list-branches
+   */
+  get ("/api/v3/repos/:owner/:repo/branches")(referrersOnly { repository =>
+    JsonFormat(JGitUtil.getBranches(
+      owner         = repository.owner,
+      name          = repository.name,
+      defaultBranch = repository.repository.defaultBranch,
+      origin        = repository.repository.originUserName.isEmpty
+    ).map { br =>
+        ApiBranchForList(br.name, ApiBranchCommit(br.commitId))
+    })
+  })
+
+  // copy code
+  private def splitPath(repository: RepositoryService.RepositoryInfo, path: String): (String, String) = {
```

Comment:
I think this method should be a member of `RepositoryService.RepositoryInfo` to avoid duplication.